From 7b61ca649245be410169e32922aca943c82a1a45 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Sun, 15 Jul 2018 21:31:45 +0200
Subject: [PATCH 5/8] TelephonyExtUtils: Add possible error codes, and return
 with them

* While we're at it, clean up comment for UICC provisioning states

Change-Id: Ifa0aec4cd8c37331f7b5231ca1147f67e1a02490
---
 .../internal/util/TelephonyExtUtils.java          | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/sdk/src/java/org/lineageos/internal/util/TelephonyExtUtils.java b/sdk/src/java/org/lineageos/internal/util/TelephonyExtUtils.java
index 59f6d10..eccf271 100644
--- a/sdk/src/java/org/lineageos/internal/util/TelephonyExtUtils.java
+++ b/sdk/src/java/org/lineageos/internal/util/TelephonyExtUtils.java
@@ -41,13 +41,18 @@ public final class TelephonyExtUtils {
 
     public static final String EXTRA_NEW_PROVISION_STATE = "newProvisionState";
 
-    // This is the list of possible values that
-    // IExtTelephony.getCurrentUiccCardProvisioningStatus() can return
+    // UICC provisioning status states
     public static final int CARD_NOT_PRESENT = -2;
     public static final int INVALID_STATE = -1;
     public static final int NOT_PROVISIONED = 0;
     public static final int PROVISIONED = 1;
 
+    // Error codes
+    public static final int SUCCESS = 0;
+    public static final int GENERIC_FAILURE = -1;
+    public static final int INVALID_INPUT = -2;
+    public static final int BUSY = -3;
+
     private boolean mNoServiceAvailable = false;
     private IExtTelephony mExtTelephony;
 
@@ -138,7 +143,7 @@ public final class TelephonyExtUtils {
                 Log.e(TAG, "Failed to get provisioning status for slotId: " + slotId, ex);
             }
         }
-        return INVALID_STATE;
+        return GENERIC_FAILURE;
     }
 
     /**
@@ -155,7 +160,7 @@ public final class TelephonyExtUtils {
                 Log.e(TAG, "Activating sub failed for slotId: " + slotId);
             }
         }
-        return -1;
+        return GENERIC_FAILURE;
     }
 
     /**
@@ -172,7 +177,7 @@ public final class TelephonyExtUtils {
                 Log.e(TAG, "Deactivating sub failed for slotId: " + slotId);
             }
         }
-        return -1;
+        return GENERIC_FAILURE;
     }
 
     /**
-- 
2.17.1

